### PR TITLE
Support GCS for storing models

### PR DESCRIPTION
**Pull request by vinvinod**
I implemented this as we were running on GCP and using GCS for storing our models. 

When running on google app engine and compute engine, the auth credentials are already set up.
For running locally or elsewhere, checkout [their client repo](https://github.com/GoogleCloudPlatform/python-docs-samples/tree/master/storage/cloud-client#authentication) for details on setting up authentication

### CODE REVIEW

**Comment by tmbo** [comment]
The only thing I am worried about with this feature is the added dependency as it is unnecessary for people not using the persistor. So I'd remove that.

Are you willing to update the PR to the current master? I think your changes should still be identical, the only issue is that we renamed `src` to `rasa_nlu`.

**Comment by vinvinod** [comment]
I wont be able to update the PR to the current master anytime soon!

**Comment by tmbo** [comment]
Alright, I'll close it for now then.

**Comment by vinvinod** [comment]
Hi,
Can we reopen this PR as I have made the necessary changes and merged with the current master.
Or should I open a new PR?

**Comment by tmbo** [comment]
Sure thing, we can reopen this one.

**Comment by angadn** [comment]
@vinvinod thanks for this Vinod! going to borrow some of this code - my approach relied on `boto 2.46.1` and https://github.com/rs/gcs-oauth2-boto-env-plugin (the ability to pass the JSON private key in via Docker environment variables was very lucrative)

File: rasa_nlu/persistor.py [hunk 1]
```diff
@@ -47,3 +78,44 @@ def fetch_and_extract(self, filename):
...
+        Persistor.__init__(self)
+        from google.cloud import storage
+        from google.cloud import exceptions
+        self.data_dir = data_dir
+        self.bucket_name = bucket_name
+        self.storage_client = storage.Client()
+
+        try:
+            self.storage_client.create_bucket(bucket_name)
+        except exceptions.Conflict, e:
```
Comment (tmbo) [inline rasa_nlu/persistor.py]: This isn't python 3 compatible (should be `exceptions.Conflict as e`)

**Pull request merged**